projects/angular/storyteller/src/app/shared/screenshot-dialog/screenshot-dialog.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, Inject, OnDestroy } from '@angular/core';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material';
+import { Subscription } from 'rxjs';
+
+import { ScreenService } from '@/core/services';
+
+@Component({
+  selector: 'screenshot-dialog',
+  templateUrl: './screenshot-dialog.component.html',
+  styleUrls: ['./screenshot-dialog.component.scss'],
+})
+export class ScreenshotDialogComponent implements OnDestroy {
+  isLoading: boolean = true;
+  imageWidth: number;
+  imageHeight: number;
+  height: number;
+  width: number;
+  subscription = new Subscription();
+
+  constructor(
+    public dialogRef: MatDialogRef<ScreenshotDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public screenshot: string,
+    private screenService: ScreenService,
+  ) {
+    this.getImageSize();
+    this.subscription = this.screenService.resizeChanges.subscribe(() => {
+      this.resize();
+    });
+  }
+
+  resize(): void {
+    this.resizeImage(
+      this.imageWidth,
+      this.imageHeight,
+      this.screenService.width - 500,
+      this.screenService.height - 100,
+    );
+  }
+
+  resizeImage(pWidth: number, pHeight: number, sWidth: number, sHeight: number): void {
+    const screenRatio: number = sWidth / sHeight;
+    const picRatio: number = pWidth / pHeight;
+    if (picRatio < screenRatio && pHeight > sHeight) {
+      this.height = sHeight;
+      this.width = pWidth * sHeight / pHeight;
+    }
+    if (picRatio >= screenRatio && pWidth > sWidth) {
+      this.width = sWidth;
+      this.height = pHeight * sWidth / pWidth;
+    }
+    this.isLoading = false;
+  }
+
+  private getImageSize(): void {
+    const img = new Image();
+    img.onload = () => {
+      this.imageWidth = img.width;
+      this.imageHeight = img.height;
+      this.resize();
```

Comment:
It's an example of a dialog component. Also you can see how to work with image ratio here.
At first you need to get image size and size of image frame (it's screen size here).
Then to count a ratio.